tools/crm_mon.c

```diff
@@ -125,7 +125,7 @@ struct {
 static void clean_up_connections(void);
 static crm_exit_t clean_up(crm_exit_t exit_code);
 static void crm_diff_update(const char *event, xmlNode * msg);
-static gboolean mon_refresh_display(gpointer user_data);
+static int mon_refresh_display(gpointer user_data);
```

Comment:
Hmm actually it looks like our mainloop functions are the ones that are wrong. glib defines GSourceFunc as returning gboolean. It doesn't look like glib uses -1. Unfortunately our mainloop functions are public API, so we can't change them right now, so this change is actually the least worst thing to do. I guess we don't get compilation warnings from g_set_source_callback() because gboolean and int are equivalent ...

As an aside, raising the required version of glib for the 2.1.0 release allows us to start using G_SOURCE_CONTINUE and G_SOURCE_REMOVE instead of 0/1/TRUE/FALSE.

Reply:
I'll modify the commit message to be more correct given reality.  And, the G_SOURCE_CONTINUE/REMOVE thing sounds like something that should be done in a separate PR since it'll also touch programs beyond crm_mon.

Reply:
Incidentally, almost all the function we add with mainloop_add_trigger return a gboolean.  Only a handful of them return an int.